Assets/Plugins/iOS/Shopify/Cart+ApplePay.mm

```diff
@@ -0,0 +1,94 @@
+//
+//  Cart+ApplePay.mm
+//  UnityBuySDK
+//
+//  Created by Shopify.
+//  Copyright © 2017 Shopify Inc. All rights reserved.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+#import "Cart+ApplePay.h"
+#import "ProductName-Swift.h"
+
+ApplePayEventDispatcher *dispatcher;
+
+NSArray *ItemsForString(NSString *jsonString) {
+    NSString *dataKey = @"Items";
+    NSError *error = nil;
+    NSDictionary *itemsJson = [NSJSONSerialization JSONObjectWithData:[jsonString dataUsingEncoding:NSUTF8StringEncoding]
+                                                              options:kNilOptions
+                                                                error:&error];
+    
+    if (error == nil && itemsJson != nil && [itemsJson objectForKey:dataKey] != nil) {
+        return itemsJson[dataKey];
+    } else {
+        return nil;
+    }
+}
+
+
+bool _CanCheckoutWithApplePay() {
+    return [PaymentSession canMakePayments];
+}
+
+bool _CanShowApplePaySetup() {
+    return [PaymentSession canShowSetup];
+}
+
+void _ShowApplePaySetup() {
+    [PaymentSession showSetup];
+}
+
+bool _CreateApplePaySession(const char *merchantID, const char *countryCode, const char *currencyCode, bool requiringShipping, const char *unityDelegateObjectName, const char *serializedSummaryItems, const char *serializedShippingMethods) {
+    
+    NSString *itemsString = [NSString stringWithUTF8String:serializedSummaryItems];
+    NSArray *itemJsons = ItemsForString(itemsString);
+    
+    NSString *shippingString = [NSString stringWithUTF8String:serializedShippingMethods];
+    NSArray *shippingJsons = ItemsForString(shippingString);
+    
+    if (itemJsons == nil) {
+        return false;
```

Comment:
If the SDK is responsible for all values being serialized, it can guarantee that deserialization will always succeed. We shouldn't return "won't ever happen" errors just because a framework indicates that errors are a possibility.